sdk/python/kubeflow/metadata/metadata.py

```diff
@@ -14,294 +14,335 @@
 
 import datetime
 import json
-from kubeflow.metadata import openapi_client
-from kubeflow.metadata.openapi_client import Configuration, ApiClient, MetadataServiceApi
-
-"""
-This module contains the Python API for logging metadata of machine learning
+import ml_metadata
+from ml_metadata.metadata_store import metadata_store
+from ml_metadata.proto import metadata_store_pb2 as mlpb
+from typing import Any, List, Mapping, Optional, TypeVar, Union
+"""This module contains the Python API for logging metadata of machine learning
 workflows to the Kubeflow Metadata service.
 """
 
-WORKSPACE_PROPERTY_NAME = '__kf_workspace__'
-RUN_PROPERTY_NAME = '__kf_run__'
-ALL_META_PROPERTY_NAME = '__ALL_META__'
+# TypeArtifact is the type Model, Dataset, Metrics or any class that has
+# class variable ARTIFACT_TYPE_NAME of string and a serialization() method that
+# returns a metadata_store_pb2.Artifact.
+TypeArtifact = TypeVar('TypeArtifact', 'Model', 'DataSet', 'Metrics', Any)
+
+_WORKSPACE_PROPERTY_NAME = '__kf_workspace__'
+_RUN_PROPERTY_NAME = '__kf_run__'
+_ALL_META_PROPERTY_NAME = '__ALL_META__'
+
+
+class Store(object):
+  """Metadata Store that connects to the Metadata gRPC service."""
+
+  def __init__(self,
+               grpc_host: str = "metadata-grpc-service.kubeflow",
+               grpc_port: int = 8080,
+               root_certificates: Optional[str] = None,
+               private_key: Optional[str] = None,
+               certificate_chain: Optional[str] = None):
+    """
+    Args:
+      grpc_host: Required gRPC service host, e.g."metadata-grpc-service.kubeflow".
+      grpc_host: Required gRPC service port.
+      root_certificates: Optional SSL certificate for secure connection.
+      private_key: Optional private_key for secure connection.
+      certificate_chain: Optional certificate_chain for secure connection.
+    """
+    config = mlpb.MetadataStoreClientConfig()
+    config.host = grpc_host
+    config.port = grpc_port
+    if root_certificates or private_key or certificate_chain:
+      config.ssl_config = config.SSLConfig()
```

Comment:
Should this be mlpb.MetadataStoreClientConfig.SSLConfig()? Maybe Protobuf support referencing the class through the instance.

Reply:
It turns out this SubMessage ssl_config field is automatically initialized with properties with None value. So I removed this line.

But I found bug of MLMD when writing unit test:

```
 File "/Users/zhenghui/kubeflow/metadata/.env/lib/python3.6/site-packages/ml_metadata/metadata_store/metadata_store.py", line 103, in _get_channel
    certificate_chain)
  File "/Users/zhenghui/kubeflow/metadata/.env/lib/python3.6/site-packages/grpc/__init__.py", line 1593, in ssl_channel_credentials
    certificate_chain))
  File "src/python/grpcio/grpc/_cython/_cygrpc/credentials.pyx.pxi", line 133, in grpc._cython.cygrpc.SSLChannelCredentials.__cinit__
TypeError: expected certificate to be bytes, got <class 'str'>
```

gRPC requires all three fields to be bytes (https://grpc.github.io/grpc/python/grpc.html#grpc.ssl_channel_credentials)

But MLMD pass them in as string.

The fix is outside the scope of this PR.

Reply:
Would you mind creating a bug for this?

Reply:
https://github.com/google/ml-metadata/issues/22

Reply:
Thanks

Reply:
@gaoning777 , shall we address this along with the python client grpc server test?

Reply:
SGTM.